### PR TITLE
Operator upgrade (from v8) with Istio Enabled domain

### PR DESCRIPTION
**Pull request by anpanigr**
The PR covers the following change

(a) Refactor the code to add a UpgradeUtil/IstioUtil class to be used in any Operator Version Upgrade scenario.
(b) Add a new Integration test class to ((i) create Domain Resource with Istio using a checked-in domain.yaml (ii) upgrade the domain to main and see if the web request still routes thru istio ingress port.

https://build.weblogick8s.org:8443/job/wko-kind-nightly-seqential/739 (entire sequential run) 
https://build.weblogick8s.org:8443/job/wko-kind-nightly-seqential/743 (entire sequential run) 
https://build.weblogick8s.org:8443/job/wko-kind-nightly-seqential/745/  (entire sequential run) 
https://build.weblogick8s.org:8443/job/wko-oke-nightly-seqential/29/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    upgradeOperatorToCurrent(opNamespace,domainNamespace,domainUid);
+    verifyPodsNotRolled(domainNamespace, pods);
+    // Re check the istio Service After Upgrade
+    checkIstioService();
+  }
+
+  private void checkIstioService() {
+    // We can not verify Rest Management console thru Adminstration NodePort
+    // in istio, as we can not enable Adminstration NodePort
+    logger.info("Verifying Istio Service @IngressPort [{0}]", istioIngressPort);
```
Comment (marinakog) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java:215]: typo : Administration

**Review by marinakog — CHANGES_REQUESTED** [review]
please run oke-sequential Jenkins job against your change. ( in the configuration switch to your branch name, provide **/testnameclass for IT_NAME

**Review by sankarpn — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+import static oracle.weblogic.kubernetes.utils.UpgradeUtils.upgradeOperatorToCurrent;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Install a released version of Operator from GitHub chart repository.
+ * Create a domain using Model-In-Image model with a dynamic cluster.
+ * Configure Itsio on the domain resource with v8 schema
+ * Make sure the console is acessible thru istio ingress port
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    assertDoesNotThrow(() -> addLabelsToNamespace(domainNamespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(opNamespace,labelMap));
+  }
+
+  /**
+   * Does some initialization of logger, conditionfactory, etc common
+   * to all test methods.
+   */
+  @BeforeAll
+  public static void init() {
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    logger = getLogger();
+  }
+
+  /**
+   * Istio Domain upgrade from Operator v3.4.1 to current.
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+ * Make sure the console is acessible thru istio ingress port
+ * Upgrade operator with current Operator image build from current branch.
+ * Make sure the console is acessible thru istio ingress port
+ */
+@DisplayName("Operator upgrade tests with Istio")
+@IntegrationTest
+@Tag("oke-sequential")
+class ItOperatorIstioUpgrade {
+
+  public static final String OLD_DOMAIN_VERSION = "v8";
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+  }
+
+  /**
+   * Cleanup Kubernetes artifacts in the namespaces used by the test and
+   * delete CRD.
+   */
+  @AfterEach
+  public void tearDown() {
+    if (!SKIP_CLEANUP) {
+      CleanupUtil.cleanup(namespaces);
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorIstioUpgrade.java]: done

**Review by marinakog — APPROVED** [review]

**Review by vanajamukkara — APPROVED** [review]

**Review by sankarpn — APPROVED** [review]
lgtm

**Pull request merged**